Modules/_tkinter.c

```diff
@@ -289,13 +289,15 @@ static PyThreadState *tcl_tstate = NULL;
       if(tcl_lock)PyThread_acquire_lock(tcl_lock, 1); \
       tcl_tstate = tstate; }
 
-#define CHECK_TCL_APPARTMENT \
-    if (((TkappObject *)self)->threaded && \
-        ((TkappObject *)self)->thread_id != Tcl_GetCurrentThread()) { \
-        PyErr_SetString(PyExc_RuntimeError, \
-                        "Calling Tcl from different apartment"); \
-        return 0; \
-    }
+#define CHECK_TCL_APPARTMENT(op) \
+    do { \
+        TkappObject *app = TkappObject_CAST(op); \
+        if (app->threaded && app->thread_id != Tcl_GetCurrentThread()) { \
+            PyErr_SetString(PyExc_RuntimeError, \
+                            "Calling Tcl from different apartment"); \
+            return 0; \
+        } \
+    } while (0)
```

Comment:
Could you move this after the `TkappObject_CAST` define, now that it's used here?

Reply:
Actually, I've observed that `CHECK_TCL_APPARTMENT` is only used by clinic functions, so there's no need for a cast. I'm not sure what's the best to do here. An alternative would be to just forget the cast.

Reply:
Would it work to put the meat into an inline function that takes `TkappObject*`, and turn the macro into `{if(check_tcl_apartment(app) < 0) return 0;}`?

Reply:
Yes, I think it's better and more robust.